hack/update_submodules.sh

```diff
@@ -4,6 +4,9 @@
 
 set -euo pipefail
 
+# the release tag to build kube-proxy from
+KUBE_PROXY_TAG=v1.30.0
```

Comment:
can we get this from Makefile, so we only maintain it in one place?

Reply:
The value in the Makefile value is populated by the submodule script (so it can be used in a make target). We only have to maintain it one place currently, which is the submodule script.

Another way to do this would be to create a config file e.g. `build.env` to hold the variable and then import it into the shell script with `source build.env` and also into the makefile with `include build.env`. I decided against this as it is just one variable, but as future work we could explore doing this for all shared variables across makefile/dockerfiles/hack scripts.